scripts/ci-entrypoint.sh

```diff
@@ -79,28 +79,22 @@ select_cluster_template() {
         # shellcheck source=scripts/ci-build-kubernetes.sh
         source "${REPO_ROOT}/scripts/ci-build-kubernetes.sh"
         export CLUSTER_TEMPLATE="test/dev/cluster-template-custom-builds.yaml"
-    elif [[ -n "${CI_VERSION:-}" ]] || [[ -n "${USE_CI_ARTIFACTS:-}" ]] || [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]]; then
+    elif [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]] || [[ -n "${CI_VERSION:-}" ]]; then
         # export cluster template which contains the manifests needed for creating the Azure cluster to run the tests
-        GOPATH="$(go env GOPATH)"
-        if ls "${GOPATH}"/src/k8s.io/kubernetes; then
-            KUBERNETES_BRANCH="$(cd "${GOPATH}/src/k8s.io/kubernetes" && git rev-parse --abbrev-ref HEAD)"
-        fi
-        if [[ "${KUBERNETES_BRANCH:-}" =~ "release-" ]]; then
-            CI_VERSION_URL="https://dl.k8s.io/ci/latest-${KUBERNETES_BRANCH/release-}.txt"
-        elif [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]]; then
+        export CLUSTER_TEMPLATE="test/ci/cluster-template-prow-ci-version.yaml"
+        if [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]]; then
```

Comment:
we can add another elif here if we want to support `stable` versions and resolve to `CI_VERSION_URL="https://dl.k8s.io/release/${KUBERNETES_VERSION}.txt`

Reply:
actually for stable- versions we don't need the ci-version template since it will just resolve to the latest released patch version, example https://storage.googleapis.com/kubernetes-release/release/stable-1.23.txt is v1.23.7 right now, so it needs to be outside the if statement that sets the template to cluster-template-prow-ci-version.yaml.

We can do this in a separate PR too.

Reply:
yeah, you're right, see my latest change

I think this is a simple enough addition that is makes sense to incorporate it in here